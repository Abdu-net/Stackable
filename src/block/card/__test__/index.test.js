import { edit } from '../index'

describe( 'Team Member', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				mediaID: 1,
				mediaURL: 'http://www.gambit.ph/',
				heading: 'Ben Adams',
				tagline: 'Ben is the head of our small team',
				des: 'Ben is the head of our small team. He loves walking his dog, Walter, when he has some free time.',
				headingColor: 'string',
				taglineColor: 'string',
				desColor: 'string',
				buttonURL: 'http://www.gambit.ph/',
				buttonText: 'Button',
				buttonColor: '#2091e1',
				buttonTextColor: '#ffffff',
				size: 'normal',
				cornerButtonRadius: 4,
				contentAlign: 'left',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				mediaID: 1,
				mediaURL: 'http://www.gambit.ph/',
				heading: 'Ben Adams',
				tagline: 'Ben is the head of our small team',
				des: 'Ben is the head of our small team. He loves walking his dog, Walter, when he has some free time.',
				headingColor: 'string',
				taglineColor: 'string',
				desColor: 'string',
				buttonURL: 'http://www.gambit.ph/',
				buttonText: 'Button',
				buttonColor: '#2091e1',
				buttonTextColor: '#ffffff',
				size: 'normal',
				cornerButtonRadius: 4,
				contentAlign: 'left',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
