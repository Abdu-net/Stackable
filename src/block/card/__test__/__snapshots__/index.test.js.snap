// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card block saved block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-card ugb-card ugb-card--design-design ugb--shadow-1 className"
    style={
      Object {
        "backgroundColor": "#123456",
        "borderRadius": 6,
      }
    }
  >
    <div
      className="ugb-card__image-container"
      data-src="mediaURL"
      style={
        Object {
          "backgroundImage": "url(mediaURL)",
          "borderRadius": undefined,
          "textAlign": "contentAlign",
        }
      }
    />
    <Unknown
      className="ugb-card__title"
      format="string"
      style={
        Object {
          "color": "#123456",
          "textAlign": "contentAlign",
        }
      }
      tagName="h4"
      value="heading"
    />
    <Unknown
      className="ugb-card__tagline"
      format="string"
      style={
        Object {
          "color": "#123456",
          "textAlign": "contentAlign",
        }
      }
      tagName="p"
      value="tagline"
    />
    <Unknown
      className="ugb-card__description"
      format="string"
      style={
        Object {
          "color": "#123456",
          "textAlign": "contentAlign",
        }
      }
      tagName="p"
      value="des"
    />
    <Unknown
      align="contentAlign"
      backgroundColor="#123456"
      borderRadius={2}
      color="#123456"
      design="buttonDesign"
      icon="buttonIcon"
      newTab={true}
      size="size"
      text="buttonText"
      url="buttonURL"
    />
  </div>
</BlockContentProvider>
`;

exports[`Card block saved default block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-card ugb-card ugb--shadow-3"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 12,
      }
    }
  >
    <Unknown
      className="ugb-card__title"
      format="string"
      style={
        Object {
          "color": undefined,
          "textAlign": "left",
        }
      }
      tagName="h4"
      value="Title for This Block"
    />
    <Unknown
      className="ugb-card__tagline"
      format="string"
      style={
        Object {
          "color": undefined,
          "textAlign": "left",
        }
      }
      tagName="p"
      value="Subtitle for this block"
    />
    <Unknown
      className="ugb-card__description"
      format="string"
      style={
        Object {
          "color": undefined,
          "textAlign": "left",
        }
      }
      tagName="p"
      value="Description for this block. Use this space for describing your block. Any text will do. Description for this block. You can use this space for describing your block. Description for this block. Use this space for describing your block. Any text will do. Description for this block. You can use this space for describing your block."
    />
    <Unknown
      align="left"
      borderRadius={4}
      color="#ffffff"
      design="basic"
      newTab={false}
      size="normal"
      text="Button text"
      url=""
    />
  </div>
</BlockContentProvider>
`;
